### PR TITLE
Updates for exception handling

### LINKED ISSUE #104: Outline Phoneme Factory commands for various inputs
**Is this related to a problem? Please describe.**
The Rhubarb tool's commands need to change depending on the user inputs. In preparation for when we receive these inputs, there should be a few if-statements to adjust the commands.

**Additional Context**
The following options will need to be specified for the commands:
- input file path
- language specifies which recognizer to use
- export format
- dialog text file path
- mouth shapes (6 basic of 9 extended)
- output file path

### LINKED ISSUE #106: Phoneme Factory exception handling
**Is this related to a problem? Please describe.**
Exception handling is required to ensure the Rhubarb tool is present, and that appropriate inputs are received, and that if they are files they are not empty.

Exception handling will need to check:
- input audio file in .wav format
- user-selected language 
- user dialog text file
- output file with phoneme timings

### PR DESCRIPTION
**Pull request by vnajack**
- Added wav corruption check with module
- `check_files()` now has promises that are not void, so the functions run in order
- Updated function names and added comments for readability
- Tested with files for peer testing, but chipmunk, cursed, reverse, and signalloss all still run in Rhubarb without error, so I didn't add exception handling for it because the files were still valid
- closes #106 
- closes #104

### CODE REVIEW

**Comment by LoganParker** [comment]
Forever jealous of the fact your issues show up with the name when you put them in

File: speaking-portal-project/src/phonemeFactory/rhubarb.ts [hunk 1]
```diff
@@ -37,11 +37,17 @@ export async function rhubarbProcessor(selected_language: string, audio_file_nam
     const rhubarbProc = spawnSync('./rhubarb', args)
 
     if (rhubarbProc.stderr) {
-        //throw Error(`${rhubarbProc.stderr}`)
+        // Rhubarb returns this as an error but it's just a status message, so do NOT throw this as an error
+        if(!rhubarbProc.stderr.includes(`Generating lip sync data for ${audio_file_name}.`)){
```
Comment (LoganParker) [inline speaking-portal-project/src/phonemeFactory/rhubarb.ts:41]: simple fix! Interesting that rhubarb & ffmpeg both throw return outputs as errors

**Review by LoganParker — APPROVED** [review]
These changes look great! Thanks for getting these done and fixing that bug!

File: speaking-portal-project/src/test/file_check.ts [hunk 2]
```diff
@@ -1,6 +1,8 @@
 import { chdir, cwd } from 'node:process'
 import fs from 'fs'
 
+var wavFileInfo = require('wav-file-info')
```
Comment (edouarde1) [inline speaking-portal-project/src/test/file_check.ts:4]: ha! This was the module fix that took a bit to realize. I wonder why 'require' statements are still needed for some modules. Will have to look into that for future

File: speaking-portal-project/src/test/file_check.ts [hunk 3]
```diff
@@ -43,17 +51,13 @@ export function checkFileExistence(file: string) {
...
      */
     fs.access(file, fs.constants.F_OK, (err) => {
-        // TODO: should we separate these checks?
-        if (err) {
-            throw Error(`FileNotFound`)
-        } else {
-            console.log(`${file} exists`)
-            return true
-        }
+        if (err) throw Error(`FileNotFound`)
```
Comment (edouarde1) [inline speaking-portal-project/src/test/file_check.ts:54]: This will be the error to catch in unit testing, I will keep that in mind

File: speaking-portal-project/src/test/file_check.ts [hunk 4]
```diff
@@ -43,17 +51,13 @@ export function checkFileExistence(file: string) {
...
     })
+    console.log(`${file} exists`)
+    return true
 }
 
-export function checkFileReadAccess(file: string) {
+export async function isFileReadable(file: string) {
```
Comment (edouarde1) [inline speaking-portal-project/src/test/file_check.ts:60]: like the name change

File: speaking-portal-project/src/test/file_check.ts [hunk 5]
```diff
@@ -64,45 +68,65 @@ export function checkFileReadAccess(file: string) {
...
+export async function isWavFileValid(file: string) {
+    /**
+     * Checks if the specified .wav file is corrupted
+     *
+     * @remarks
+     * Ensure checkFileExistence(), checkFileReadAccess(), and checkWavFile() are completed prior to this check
+     *
+     * @param file - The name of the file to check
+     */
+    wavFileInfo.infoByFilename(file, function(err:any, info:any){
```
Comment (edouarde1) [inline speaking-portal-project/src/test/file_check.ts:103]: Future we could somehow type out the error 'type', rather than using 'any' here. Strange this module as not imported types for this function

File: speaking-portal-project/src/test/file_check.ts [hunk 6]
```diff
@@ -64,45 +68,65 @@ export function checkFileReadAccess(file: string) {
...
 
-    // TODO: should we check if it's corrupted? how?
-
-    // TODO: check if text file is empty
+    // Check if text file is readable in utf8 format and check if empty
     fs.readFile(file, 'utf8', (err, data) => {
-        if (err) throw Error(`Unable to read ${file}`)
+        if (err) throw Error(`UnableToReadTextFile`)
+        console.log(`Text file ${file} is valid`)
         if (data.length > 0) {
```
Comment (edouarde1) [inline speaking-portal-project/src/test/file_check.ts:123]: Will we ever need to add a threshold for this value ?

- if the txt file only has 5 chars ? or 10 ?  I don't think it would really matter, but just a thought

File: speaking-portal-project/src/test/test.ts [hunk 7]
```diff
@@ -1,20 +1,20 @@
 import {test,expect} from '@jest/globals'
-import { checkFileExistence,checkFileReadAccess,checkWavFile,checkTextFile } from "./file_check";
+import { doesFileExist,isFileReadable,isWavFile,isTextFileValid } from "./file_check";
 
 
 
-test('File Not Found', () => { 
-    expect(() =>checkFileExistence('test.wav')).toThrow('FileNotFound')
+test('File Not Found', () => {
+    expect(() =>doesFileExist('test.wav')).toThrow('FileNotFound')
```
Comment (edouarde1) [inline speaking-portal-project/src/test/test.ts:7]: sweet thanks adjusting the unit test functions! 

I will work on getting these to run once the PR is merged. The files in the inputs rn are fake files. I will adjust to make sure we are testing the correct testing file for each
edge case.

**Review by edouarde1 — APPROVED** [review]
Great job Veronica! This error handling functions look solid. Just a couple comments I left

File: speaking-portal-project/src/phonemeFactory/rhubarb.ts (hunk 1 above)
Comment (vnajack) [reply speaking-portal-project/src/phonemeFactory/rhubarb.ts:41]: Yeah, I'm not sure why they throw these outputs as errors, but at least we recognize that!!

File: speaking-portal-project/src/test/file_check.ts (hunk 2 above)
Comment (vnajack) [reply speaking-portal-project/src/test/file_check.ts:4]: it might be due to the age of this module, but I would also like to look further into it.

File: speaking-portal-project/src/test/file_check.ts (hunk 5 above)
Comment (vnajack) [reply speaking-portal-project/src/test/file_check.ts:103]: If you refactor the catches with `err:Error`, let me know how it goes or please share your screen when you do it! I feel like this change might require more than we think...

File: speaking-portal-project/src/test/file_check.ts (hunk 6 above)
Comment (vnajack) [reply speaking-portal-project/src/test/file_check.ts:123]: I think we should stick with whether the text file is empty or not. In Japanese, they use Kanji which can encompass an entire word, so you could write a sentence with 5 characters.

File: speaking-portal-project/src/test/test.ts (hunk 7 above)
Comment (vnajack) [reply speaking-portal-project/src/test/test.ts:7]: Sounds good, thanks! Let me know if you'd like any help with that, or if you find another test I should include.
